pkg/segment/aggregations/streamstats.go

```diff
@@ -0,0 +1,863 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package aggregations
+
+import (
+	"container/list"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/siglens/siglens/pkg/common/dtypeutils"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	putils "github.com/siglens/siglens/pkg/utils"
+)
+
+func GetBucketKey(record map[string]interface{}, groupByRequest *structs.GroupByRequest) (string, error) {
+	bucketKey := ""
+	for _, colName := range groupByRequest.GroupByColumns {
+		val, ok := record[colName]
+		if !ok {
+			return "", fmt.Errorf("GetBucketKey: Error: column: %v not found in the record", colName)
+		}
+		bucketKey += fmt.Sprintf("%v_", val)
+	}
+	return bucketKey, nil
+}
+
+func InitRunningStreamStatsResults(defaultVal float64) *structs.RunningStreamStatsResults {
+	return &structs.RunningStreamStatsResults{
+		Window:          list.New().Init(),
+		CurrResult:      defaultVal,
+		SecondaryWindow: list.New().Init(),
+	}
+}
+
+func InitRangeStat() *structs.RangeStat {
+	return &structs.RangeStat{
+		Min: math.MaxFloat64,
+		Max: -math.MaxFloat64,
+	}
+}
+
+func PerformGlobalStreamStatsOnSingleFunc(ssOption *structs.StreamStatsOptions, ssResults *structs.RunningStreamStatsResults, measureAgg utils.AggregateFunctions, colValue interface{}) (float64, bool, error) {
```

Comment:
It means whether the value exists or not. Currently in Splunk when the current is false you check for the values present before the current event. Due to reset or it is the first event, you cannot compute the value because the values are not there. So, you'll need to show a blank field. However, for count it displays 0 cause it has some meaning.